### ISSUE TITLE
Destroys ref prop passed to children

### ISSUE THREAD

**Issue by natew** [joshwcomeau/react-flip-move#140]
```js
        <FlipMove>
            {[1, 2, 3, 4].map(x => <Test key={x} ref={console.log} />)}
          </FlipMove>
```

Ref wont callback here at all. But if you move it outside of map it works just fine!

**Comment by joshwcomeau** [comment]
Hm, you shouldn't be able to assign refs on children passed to FlipMove anyway - FlipMove clones the set of children you provide and overwrites the ref.

What is it that you're trying to do exactly?

**Comment by natew** [comment]
I need to call methods on the children to do various things with them. An
example is focusing their text input on the item you click.

**Comment by joshwcomeau** [comment]
Yeah, so as long as the ref isn't on the direct child, everything should be fine.

```js
class YourForm extends Component {
  render() {
    return (
      <FlipMove>
        {[1,2,3,4].map(id => (
          <div>
            <input ref={...} />
          </div>
        )}
      </FlipMove>
    )
  }
}
```

You can (and maybe should) also extract the input into its own component.

```js
class YourInput extends Component {
  render() {
    return <input ref={...} />;
  }
}

class YourForm extends Component {
  render() {
    return (
      <FlipMove>
        {[1,2,3,4].map(id => <YourInput key={id} />}
      </FlipMove>
    )
  }
}
```

**Comment by natew** [comment]
Yea the input is in a component. I am calling focus on the parent
component. But good point I guess I can nest it, confusing to figure out
though would be good to document.

**Comment by joshwcomeau** [comment]
Yeah, I can see how that'd be confusing. We could add it to the "Gotchas", although I don't know how much good that does; it isn't a place people typically check when experiencing an issue.

Maybe we can add a console warning if FlipMove detects a pre-existing ref on a supplied component.

Would be a great contribution if you or anyone else reading this wants to take it on :)

**Comment by natew** [comment]
Yep that or it could take a function as children possibly which I think
avoids that problem as well.

**Comment by joshwcomeau** [comment]
@jahooma interesting! That does seem like it'd work.

My time is stretched far too thin right now, but if you (or anyone else reading this) wanted to open a PR with that change, I'd happily merge it (just be sure to test that it works in the local storybook).

Also thanks for sharing throne.live, I always love seeing where Flip Move gets used :)
